### PR TITLE
Add detail rbacassessment,infraassessment info prometheus Metrics

### LINKED ISSUE #1310: Expose Detailed Reporting as prometheus Metrics
it is required to  enhance the metrics available in Prometheus for the following reports:
-  RbacAssessmentReport 
-  InfraAssessmentReport 
-  ClusterComplianceReport. 

- The current metrics provided by Prometheus lack sufficient detail, which limits the effectiveness of monitoring and analysis.

```[tasklist]
### Tasks
- [ ] https://github.com/aquasecurity/trivy-operator/issues/1148
```

### PR DESCRIPTION
**Pull request by geoffrey1330**
## Description
Add detail rbacassessment,infraassessment info prometheus Metrics

## Related issues
- Close #1310 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@geoffrey1330 see my comment above

**Comment by chen-keinan** [comment]
@geoffrey1330 lgtm , one comment please update [metric docs](https://github.com/aquasecurity/trivy-operator/blob/main/docs/tutorials/integrations/metrics.md)

**Comment by chen-keinan** [comment]
@geoffrey1330 you'll need to run `make generate-all` and push changes to fix the `verify code` issue

**Comment by chen-keinan** [comment]
@geoffrey1330  you'll have to rebase with upstream and resolve conflict

**Comment by geoffrey1330** [comment]
> @geoffrey1330 you'll need to run `make generate-all` and push changes to fix the `verify code` issue

@chen-keinan Verify code is still failing after i ran the command

File: deploy/helm/generated/role.yaml [hunk 1]
```diff
@@ -108,6 +108,14 @@ rules:
...
   - list
   - watch
+- apiGroups:
+  - apps.openshift.io
+  resources:
+  - deploymentconfigs
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (chen-keinan) [inline deploy/helm/generated/role.yaml:118]: you rebase was done wrongly, this should not be part of your code

**Comment by chen-keinan** [comment]
see my comment , you rebase was done wrongly , your PR include changes from other PRs

**Pull request closed without merging**